### PR TITLE
Add missing_ok option to the SQL functions for reading files.

### PR DESCRIPTION
**Pull request by ashwinstar**
This is cherry-picking upstream commit to get the functions required by 9.5 `pg_rewind`. Since the dependency is not much, opted to pull the function. Another option is to hack 9.5 pg_rewind similar to outside postgresql repo version for 9.4. It creates the functions only for pg_rewind but seems very hacky and this is better approach.

```
Author: Heikki Linnakangas <heikki.linnakangas@iki.fi>
Date:   Sun Jun 28 21:35:46 2015 +0300

Add missing_ok option to the SQL functions for reading files.

This makes it possible to use the functions without getting errors, if there
is a chance that the file might be removed or renamed concurrently.
pg_rewind needs to do just that, although this could be useful for other
purposes too. (The changes to pg_rewind to use these functions will come in
a separate commit.)

The read_binary_file() function isn't very well-suited for extensions.c's
purposes anymore, if it ever was. So bite the bullet and make a copy of it
in extension.c, tailored for that use case. This seems better than the
accidental code reuse, even if it's a some more lines of code.

Michael Paquier, with plenty of kibitzing by me.
```

### CODE REVIEW

**Comment by paul-guo-** [comment]
I guess we need to modify gpdb specific file src/include/catalog/pg_proc.sql for those functions?

File: src/test/regress/expected/opr_sanity.out [hunk 1]
```diff
@@ -1295,14 +1296,6 @@ WHERE a.aggfnoid = p.oid AND
 -- combine(transtype, transtype) returns transtype
 -- NOTE: use physically_coercible here, not binary_coercible, because
 -- max and min on abstime are implemented using int4larger/int4smaller.
--- start_ignore
```
Comment (hlinnaka) [inline src/test/regress/expected/opr_sanity.out]: Unrelated change, let's not remove this comment.

File: src/test/regress/expected/opr_sanity.out [hunk 2]
```diff
@@ -139,12 +139,13 @@ WHERE p1.oid < p2.oid AND
      p1.pronargs != p2.pronargs);
  oid  |      proname      | oid  |       proname       
 ------+-------------------+------+---------------------
+ 3293 | pg_read_file      | 6045 | pg_file_read
```
Comment (hlinnaka) [inline src/test/regress/expected/opr_sanity.out]: This isn't needed in the upstream. Why we need it?

**Comment by hlinnaka** [comment]
Yeah, makes sense to backport this.

File: src/test/regress/expected/opr_sanity.out (hunk 2 above)
Comment (ashwinstar) [reply src/test/regress/expected/opr_sanity.out]: Thank you for catching!!

GPDB has defined `pg_file_read()` in `pg_proc.sql` and the only difference it has from `pg_read_file()` upstream function is prodataaccess = `r` (`PRODATAACCESS_READS`) for GPDB function whereas upstream gets marked with `n` (`PRODATAACCESS_NONE`) by default.

So,

1. do we need to keep around `pg_file_read()`?
2. or can we just replace it with `pg_read_file()`? If can be replaced do we need to mark the `pg_read_file()` and other functions which has `prosrc == pg_read_file` with prodataaccess = `r`?
3. If `prodataaccess` has to be set to non-default values for upstream functions, how do we approach the changes in those cases? Comment out the code in `pg_proc.h` and write the function in `pg_proc.sql` or any other better alternatives exists?

Comment (ashwinstar) [reply src/test/regress/expected/opr_sanity.out]: @hlinnaka looking for input to the above questions.

Comment (ashwinstar) [reply src/test/regress/expected/opr_sanity.out]: Seems upstream doesn't have `pg_file_read()` and friends defined in catalog only `contrib/adminpack/adminpack--1.0.sql` has it. Greenplum have defined them in catalog.

**Comment by hlinnaka** [comment]
This was pushed already, as commit d19a7332c8. Closing.
